common/src/arch-riscv64.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * See the dyninst/COPYRIGHT file for copyright information.
+ *
+ * We provide the Paradyn Tools (below described as "Paradyn")
+ * on an AS IS basis, and do not warrant its validity or performance.
+ * We reserve the right to update, modify, or discontinue this
+ * software at any time.  We shall have no obligation to supply such
+ * updates or modifications or any other form of support to you.
+ *
+ * By your use of Paradyn, you understand and agree that we (or any
+ * other person or entity with proprietary rights in Paradyn) are
+ * under no obligation to provide either maintenance services,
+ * update services, notices of latent defects, or correction of
+ * defects for Paradyn.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#ifndef _ARCH_RISCV64_H
+#define _ARCH_RISCV64_H
+
+#include <stdint.h>
+
+// In RISC-V, instruction length can be a multiple of 2 bytes
+
+// Standard RISC-V instructions (RVI, RVA, RVM, RVF, RVD, ...) are 4 bytes
+using rvInsn_t = uint32_t;
+constexpr int RISCV_INSN_SIZE = sizeof(rvInsn_t);
+
+// Compressed instructions (RVC) are 2 bytes
+using rvcInsn_t = uint16_t;
+constexpr int RISCVC_INSN_SIZE = sizeof(rvcInsn_t);
+
+// The minimum instruction size is 2 bytes
+using rvMinInsn_t = rvcInsn_t ;
+constexpr int RISCV_MIN_INSN_SIZE = sizeof(rvMinInsn_t);
+
+// The maximum instruction size is 4 bytes
+using rvMaxInsn_t = rvInsn_t;
+constexpr int RISCV_MAX_INSN_SIZE = sizeof(rvMaxInsn_t);
+
+// The register size is 4 bytes
```

Comment:
I don't understand this comment. What register? RISC-V registers are 64-bits in RV64.

Reply:
Comments are updated.